lib/puppet/defaults.rb

```diff
@@ -2023,7 +2025,7 @@ def self.initialize_default_settings!(settings)
       :desc     => "The LDAP server.",
     },
     :ldapport => {
-      :default  => 389,
+      :default  => "389",
```

Comment:
I think it’s ok to keep this as an int and change the test, because before the settings changes, if the value was set in puppet.conf then it would have automatically been converted to an integer.

Reply:
I went to go do this and when looking for the test in question again I noticed the extent we use strings for ports (see https://github.com/puppetlabs/puppet/blob/b7c9dd387a23718ba552d07e16f32533ff343b5a/spec/unit/util/ldap/connection_spec.rb). Shall I go ahead and change those occurrences as well? (Happy to)

Reply:
I think it  makes sense to also update those places, seems like we're making an effort to get a little more strict about settings types.

Reply:
Just to make sure changing the type to `:integer` won't break anyone, I double checked that before any of our changes, the `Puppet:Util::Ldap::Connection` class is called with an integer port when using the defaults or if the port is set in puppet.conf:

```
# puppet apply user.pp --ldapuser ldap
Notice: Compiled catalog for brisker-chemise.delivery.puppetlabs.net in environment production in 0.01 seconds
connect: "ldap":389
# puppet config set ldapport 636 --section main
# puppet apply user.pp --ldapuser ldap
Notice: Compiled catalog for brisker-chemise.delivery.puppetlabs.net in environment production in 0.01 seconds
connect: "ldap":636
```

However, if `Puppet[:ldapport] = "636"`, then `Puppet[:ldapport]` used to return a string. And I'm :+1: about removing that special case behavior, specially this test: https://github.com/puppetlabs/puppet/blob/b7c9dd387a23718ba552d07e16f32533ff343b5a/spec/unit/util/ldap/connection_spec.rb#L124-L125

Instead, we should `expect(...).to receive(..., 456,...)`

Feel free to change the other places, like https://github.com/puppetlabs/puppet/blob/b7c9dd387a23718ba552d07e16f32533ff343b5a/spec/unit/util/ldap/connection_spec.rb#L74..., as passing "myport" isn't very realistic.

Reply:
Just to be clear, @joshcooper do you want the type of this to be `:integer` or `:port`?

Reply:
I set it to port, but if `:integer` is more correct for some reason, let me know.